### DISCUSSION TITLE
Help formatting MVT

### DISCUSSION THREAD

**Discussion by Joebjammin** [TerriaJS/terriajs#6381] [Q&A] [answered]
Hi All,
I am trying to display polygon MVT jn TerriaJS 8. I have not been able to get them to display. I normally do this before I try region mapping. Any help you can provide would be most appreciated. Here is my config for testing. I have tried both on-line and local repository. Attached are my MVT tiles. 
[FL_COUNTY.zip](https://github.com/TerriaJS/terriajs/files/9034951/FL_COUNTY.zip)

{
    "homeCamera": {
      "north": 31.0,
      "east": -80.0,
      "south": 24.5,
      "west": -87.6
    },
    "catalog": [
     {
        "id": "1",
        "type": "geojson",
        "name": "FL_ZCTA2018",
        "url": "test/FL_ZCTA2018.geojson"
      },
//Works with TerriaJS 7//
     /* {
        "id": "2",
        "type": "mvt",
        "url": "https://terria.tad3.org/data/s3/FL_COUNTY/{z}/{x}/{y}.pbf",
        "layer": "FL_COUNTY",
        "minimumZoom": 3,
        "maximumNativeZoom": 9,
        "fillColor": "orange",
        "lineColor": "yellow",
        "rectangle": [
          -87.6348959282059781,
          24.5210316754366957,
          -80.0313186453441432,
          31.0001954218924993
          ]
        }
//Trying with TerriaJS 8 - Online Repository//   
       {
        "id": "3",
        "type": "mvt",
        "name": "FL_COUNTY",
        "id": "2",
        "url": "https://terria.tad3.org/data/s3/FL_COUNTY/{z}/{x}/{y}.pbf",
        "layer":"FL_COUNTY",
        "lineColor": "red",
        "fillColor": "green",
        "minimumZoom":3,
        "maximumNativeZoom":9,
        "maximumZoom": 9,
        "idProperty":"GEOID",
        // "nameProperty":"Name",
        "keepOnTop": true,
        "rectangle": {
        "west": -87.6347,
        "south": 24.514909,
        "east": -80.032576,
        "north": 31.000809
        },
        },
//Trying with TerriaJS 8 - Test Directory// 
        {
        "id": "3",   
        "name": "FL_COUNTY",
        "type": "mvt",
        "url": "test/FL_COUNTY/{z}/{x}/{y}.pbf",
        "layer": "FL_COUNTY",
        "minimumZoom": 3,
        "maximumNativeZoom": 9,
        "fillColor": "orange",
        "lineColor": "yellow",
        "rectangle": [
            -87.6348959282059781,
            24.5210316754366957,
            -80.0313186453441432,
            31.0001954218924993
            ]
        },*/
    ],
    "viewerMode": "3dSmooth",
    "baseMaps": {
      "defaultBaseMapId": "basemap-positron",
      "previewBaseMapId": "basemap-positron"
    }
  }

**Comment by nf-s** [comment, marked as answer]
Hi @Joebjammin 

Currently, Terria requires that `minimumZoom` must be at least 1 - this is so we don't request more than 4 tiles when a catalog item is added to the map.

The `rectangle` property has also changed, it is now an object with `west`, `south`, `east` and `north` properties.

The following JSON works for me on the latest TerriaJS

By setting `"minimumZoom": 1,` - Terria will attempt to load vector tiles for zoom level 1 and 2 - even though they don't exist

```json
{
      "id": "3",
      "name": "FL_COUNTY",
      "type": "mvt",
      "url": "FL_COUNTY/{z}/{x}/{y}.pbf",
      "layer": "FL_COUNTY",
      "minimumZoom": 1,
      "maximumZoom": 10,
      "fillColor": "orange",
      "lineColor": "yellow",
      "rectangle": {
        "west": -87.6348959282059781,
        "south": 24.5210316754366957,
        "east": -80.0313186453441432,
        "north": 31.0001954218924993
      }
    }
```

I believe the behavior of `minimumZoom` is unchanged from version 7

### Version 8

https://github.com/TerriaJS/terriajs/blob/fc33525049a701c3d1876b9ff9fbffc1c65e4310/lib/Map/ImageryProvider/ProtomapsImageryProvider.ts#L310-L329

### Version 7

https://github.com/TerriaJS/terriajs/blob/a6a2e6464cec91578b924728cdb3e3689b8c2794/lib/Map/MapboxVectorTileImageryProvider.js#L66-L85

**Comment by Joebjammin** [comment]
Thank you. I appreciate your work!
